### PR TITLE
SER-4524 [Mobile] Search scope toggle

### PR DESCRIPTION
**Pull request by mmeller-wikia**

### CODE REVIEW

**Review by slayful — APPROVED** [review]

**Review by latata — APPROVED** [review]

File: app/components/global-navigation/search.js [hunk 1]
```diff
@@ -411,6 +414,10 @@ export default Component.extend({
 		// ENTER key
 		} else if (keyCode === 13) {
 			if (this.selectedSuggestionIndex !== -1) {
+				if (this.getScope() === 'cross-wiki') {
+					window.location.replace(this.suggestions[this.selectedSuggestionIndex].uri);
```
Comment (kasperekt) [inline app/components/global-navigation/search.js:418]: Why `replace` instead on `push`? Is it intentional?

File: app/components/global-navigation/search.js [hunk 2]
```diff
@@ -430,6 +437,15 @@ export default Component.extend({
 		return this.get('searchScope') === 'cross-wiki' ? 'cross-wiki' : 'internal';
 	},
 
+	getDefaultScope() {
+		const savedSearchScope = Cookies.get('DEFAULT_SEARCH_SCOPE');
```
Comment (lomiarz) [inline app/components/global-navigation/search.js:441]: how about just `return Cookies.get('DEFAULT_SEARCH_SCOPE') || 'internal'`

File: app/components/global-navigation/search.js [hunk 3]
```diff
@@ -430,6 +437,15 @@ export default Component.extend({
...
+		if (savedSearchScope === 'cross-wiki' || savedSearchScope === 'internal') {
```
Comment (lomiarz) [inline app/components/global-navigation/search.js:442]: did you consider to use some enum/constants for cross-wiki and internal scope ?

File: app/components/global-navigation/search.js (hunk 1 above)
Comment (mmeller-wikia) [reply app/components/global-navigation/search.js:418]: We need full reload (redirect to different wiki) and this is just the first solution that worked for me
I can't find `window.location.push` function, so if you can provide snippet i would be grateful :)

File: app/components/global-navigation/search.js (hunk 2 above)
Comment (mmeller-wikia) [reply app/components/global-navigation/search.js:441]: I would stick with this little validation. Just because different value will break nav-search

**Review by lomiarz — APPROVED** [review]

**Pull request merged**